### PR TITLE
fix for issue #403

### PR DESCRIPTION
**Pull request by dnutcracker**

### CODE REVIEW

File: library/src/main/java/com/google/android/exoplayer/upstream/AssetDataSource.java [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+        }
+      }
+      Log.i("AssetDataSource", "read(). bytesRead = " + bytesRead);
+
+      return bytesRead;
+    }
+  }
+
+  @Override
+  public void close() throws FileDataSourceException {
```
Comment (ojw28) [inline library/src/main/java/com/google/android/exoplayer/upstream/AssetDataSource.java]: Should throw AssetDataSourceException instead.

File: library/src/main/java/com/google/android/exoplayer/upstream/AssetDataSource.java [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+        throw new AssetDataSourceException(e);
+      }
+
+      if (bytesRead > 0) {
+        bytesRemaining -= bytesRead;
+        if (listener != null) {
+          listener.onBytesTransferred(bytesRead);
+        }
+      }
+      Log.i("AssetDataSource", "read(). bytesRead = " + bytesRead);
```
Comment (ojw28) [inline library/src/main/java/com/google/android/exoplayer/upstream/AssetDataSource.java]: Can we get rid of this. It's pretty spammy (guessing you had it for debugging).

**Comment by ojw28** [comment]
This looks great, thanks! Please tweak as per the suggestions. Other than that we can merge this, but you'll need to electronically sign a CLA first. See: https://github.com/google/ExoPlayer/blob/master/CONTRIBUTING.md.

File: library/src/main/java/com/google/android/exoplayer/upstream/AssetDataSource.java (hunk 2 above)
Comment (dnutcracker) [inline library/src/main/java/com/google/android/exoplayer/upstream/AssetDataSource.java]: Yep

File: library/src/main/java/com/google/android/exoplayer/upstream/AssetDataSource.java (hunk 1 above)
Comment (dnutcracker) [inline library/src/main/java/com/google/android/exoplayer/upstream/AssetDataSource.java]: Yep again

**Comment by dnutcracker** [comment]
done and done :)

**Comment by ojw28** [comment]
Thanks!

**Pull request merged**